cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java

```diff
@@ -137,13 +137,44 @@ public void testDeployVersionedApp() throws Exception {
     AppRequest<ConfigTestApp.ConfigClass> request = new AppRequest<>(
       new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()), config);
     Assert.assertEquals(200, deploy(appId, request).getStatusLine().getStatusCode());
-
-    JsonObject appDetails = getAppDetails(Id.Namespace.DEFAULT.getId(), appId.getApplication(), appId.getVersion());
+    Assert.assertEquals(500, deploy(appId, request).getStatusLine().getStatusCode());
+    Assert.assertEquals(404, getVersionedAppResponse(Id.Namespace.DEFAULT.getId(), appId.getApplication(),
+                                            "non_existing_version").getStatusLine().getStatusCode());
+    Assert.assertEquals(404, getNonVersionedAppResponse(Id.Namespace.DEFAULT.getId(),
+                                                        appId.getApplication()).getStatusLine().getStatusCode());
+
+    Id.Application appIdDefault = Id.Application.from(Id.Namespace.DEFAULT, appId.getApplication());
```

Comment:
It would be a good idea to add some comments on what the test is doing. For example over here you can say `// Deploy non-versioned app`